frontend/src/components/maps/leaflet/LayersControl/wmsHeaders.ts

```diff
@@ -0,0 +1,28 @@
+import { AxiosResponse } from 'axios';
+import CustomAxios from 'customAxios';
+import L from 'leaflet';
+
+/**
+ * Override the base leaflet WMS functionality to handle adding an auth token to the requests.
+ */
+class WmsHeaders extends L.TileLayer.WMS {
+  createTile(coords: any, done: any) {
+    const url = this.getTileUrl(coords);
+    const img = document.createElement('img');
+    const axios = CustomAxios();
+    axios.get(url, { responseType: 'blob' }).then((response: AxiosResponse) => {
```

Comment:
@asanchezr this will likely fail unless the cors policy on the geoserver is set up to allow requests from PSP. Maybe we can talk to Peter Spry about this tomorrow.